legacy/builder/phases/linker.go

```diff
@@ -63,46 +65,150 @@ func (s *Linker) Run(ctx *types.Context) error {
 	return nil
 }
 
+// CppArchive represents a cpp archive (.a). It has a list of object files
+// that must be part of the archive and has functions to build the archive
+// and check if the archive is up-to-date.
+type CppArchive struct {
```

Comment:
Is CppArchive a good name here? It is not Cpp-specific, but really just contains object files (which could be generated from C or assembly sources too)?